app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml

```diff
@@ -27,25 +28,20 @@ $paddingBottom = $block->getRatio() * 100;
 $styles = <<<STYLE
 .product-image-container-{$block->getProductId()} {
     width: {$width}px;
+    height: auto;
+    aspect-ratio: auto {$width} / {$height};
 }
 .product-image-container-{$block->getProductId()} span.product-image-wrapper {
-    padding-bottom: {$paddingBottom}%;
+    height: 100%;
+    width: 100%;
 }
-STYLE;
-//In case a script was using "style" attributes of these elements
-$script = <<<SCRIPT
-prodImageContainers = document.querySelectorAll(".product-image-container-{$block->getProductId()}");
-for (var i = 0; i < prodImageContainers.length; i++) {
-    prodImageContainers[i].style.width = "{$width}px";
-}
-prodImageContainersWrappers = document.querySelectorAll(
-    ".product-image-container-{$block->getProductId()}  span.product-image-wrapper"
-);
-for (var i = 0; i < prodImageContainersWrappers.length; i++) {
-    prodImageContainersWrappers[i].style.paddingBottom = "{$paddingBottom}%";
+@supports not (aspect-ratio: auto) { 
+    .product-image-container-{$block->getProductId()} span.product-image-wrapper {
+        padding-bottom: {$paddingBottom}%;
+        height: auto;
```

Comment:
I think `padding-bottom: N%` + `height: auto` might cause an issue when the block will be rendered with 2 x heights in some cases. What do you think?